### DISCUSSION TITLE
Attach location to the route

### DISCUSSION THREAD

**Discussion by Shayybe** [mapsforge/mapsforge#1529] [Q&A] [answered]
Ok so this might be a very noob question but i am struggling with this since im new to java. I want my navigation icon to be attached to the navigation route path such that it travels along the route in my Android nav app? i did read online where mapsforge and graphhopper are totally standalone and or not dependent on each other.......any sort of guidance or direction i would greatly appriciate it.

**Comment by devemux86** [comment]
If you mean the map center to follow the GPS location, this could be done with the Android location listener.
A simple example is in the [LocationOverlayMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/LocationOverlayMapViewer.java#L104-L110) sample.

**Reply by devemux86** [reply, marked as answer]
Follow location can set the map center on the real GPS location.
If you mean the map center to snap on the route, this is not related to the map library.

It's mathematics, better search online for how to project the GPS location on the route.
Then can set the location overlay and map center to the new found location.

**Reply by Shayybe** [reply]
ohhhhhh thannk you sooo much ill be sure to check it out. thanks
